planner/core/logical_plans.go

```diff
@@ -216,6 +216,7 @@ func (p *LogicalJoin) columnSubstitute(schema *expression.Schema, exprs []expres
 
 		p.EqualConditions[i] = newCond
 	}
+	p.cartesianJoin = len(p.EqualConditions) == 0
```

Comment:
why do we need to maintain `p.cartesianJoin` in here?

Reply:
`create table t1(a bigint, b bigint); create table(a bigint, b bigint);`
` select * from t1 where t1.a in (select t1.b + t2.b from t2)` can be rewrite to `select distinct t1.* from t1 join t2 on t1.a = t1.b + t2.b`.

the given testcase say it's a cartesian join, I need to revise it, it should not be a cartesian join IMO.

Reply:
It is not needed.

The given testcase `select * from t1 where t1.a in (select t1.b + t2.b from t2)` is said a cartesian join, it's a mistake.